cumulus_etl/common.py

```diff
@@ -32,49 +31,13 @@ def ls_resources(root, resource: str) -> list[str]:
 #
 ###############################################################################
 
-_user_fs_options = {}  # don't access this directly, use get_fs_options()
 
-
-def set_user_fs_options(args: dict) -> None:
-    """Records user arguments that can affect filesystem options (like s3_region)"""
-    _user_fs_options.update(args)
-
-
-def get_fs_options(protocol: str) -> dict:
-    """Provides a set of storage option kwargs for fsspec calls or pandas storage_options arguments"""
-    options = {}
-
-    if protocol == "s3":
-        # Check for region manually. If you aren't using us-east-1, you usually need to specify the region
-        # explicitly, and fsspec doesn't seem to check the environment variables for us, nor pull it from
-        # ~/.aws/config
-        region_name = _user_fs_options.get("s3_region")
-        if region_name:
-            options["client_kwargs"] = {"region_name": region_name}
-
-        # Assume KMS encryption for now - we can make this tunable to AES256 if folks have a need.
-        # But in general, I believe we want to enforce server side encryption when possible, KMS or not.
-        options["s3_additional_kwargs"] = {
-            "ServerSideEncryption": "aws:kms",
-        }
-
-        # Buckets can be set up to require a specific KMS key ID, so allow specifying it here
-        kms_key = _user_fs_options.get("s3_kms_key")
-        if kms_key:
-            options["s3_additional_kwargs"]["SSEKMSKeyId"] = kms_key
-
-    return options
-
-
-def open_file(path: str, mode: str):
-    """A version of open() that handles remote access, like to S3"""
-    # Grab protocol if present
-    parsed = urlparse(path)
-    protocol = parsed.scheme or "file"  # assume local if no obvious scheme
-
-    # We pass auto_mkdir because on some backends (like S3), we may not have permissions that fsspec might want,
-    # like CreateBucket. We elsewhere call Root.makedirs as needed.
-    return fsspec.open(path, mode, encoding="utf8", auto_mkdir=False, **get_fs_options(protocol))
```

Comment:
auto_mkdir logic here does not need to be replicated - it's on by default only for the main entry points `fsspec.open` and `fsspec.open_files` -- actual filesystems have it off by default. :shrug: